backstopper-core/src/main/java/com/nike/backstopper/handler/ApiExceptionHandlerBase.java

```diff
@@ -372,11 +376,20 @@ protected boolean shouldLogStackTrace(
         int statusCode, Collection<ApiError> filteredClientErrors, Throwable originalException,
         Throwable coreException, RequestInfoForLogging request
     ) {
-        // By default, 4xx should *not* log stack trace. Everything else should.
-        //noinspection RedundantIfStatement
-        if (statusCode >= 400 && statusCode < 500)
+        if(coreException instanceof ApiException) {
```

Comment:
I don't think this logic is correct. If `stackTraceLoggingBehavior` is null or `DEFER_TO_DEFAULT_BEHAVIOR` then we'd want the status code checks to be done. We only want to short circuit return true or false when `stackTraceLoggingBehavior` equals `FORCE_STACK_TRACE` or `FORCE_NO_STACK_TRACE`.

Reply:
i see, my bad, will fix this. Thanks!